src/pages/index.js

```diff
@@ -166,44 +227,99 @@ addButton.addEventListener("click", () => openModal(addModal));
+//handles edit profile submit    ---- FIX RELOAD NAME
 function handleEditProfileSubmit(evt) {
   evt.preventDefault();
-  profileNameEl.textContent = editProfileNameInput.value;
-  profileDescriptionEl.textContent = editProfileDescriptionInput.value;
-  closeModal(editModal);
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Saving...");
+
+  api
+    .editUserInfo({
+      name: editProfileNameInput.value,
+      about: editProfileDescriptionInput.value,
+    })
+    .then((user) => {
+      profileNameEl.textContent = user.name;
+      profileDescriptionEl.textContent = user.about;
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+      closeModal(editModal);
+    });
 }
 
 editProfileForm.addEventListener("submit", handleEditProfileSubmit);
 
+addPostForm.addEventListener("submit", handleAddCardSubmit);
+
 //handleAddPostsubmit
 function handleAddCardSubmit(evt) {
   evt.preventDefault();
 
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Saving...");
+
   const inputValues = {
     name: postCaptionInput.value,
     link: postImageInput.value,
   };
 
-  const cardElement = getCardElement(inputValues);
-  cardsList.prepend(cardElement);
-
-  addPostForm.reset();
-  resetValidation(addPostForm, settings);
-  disableButton(cardSubmitBtn); //added*****************
-  closeModal(addModal);
+  api
+    .addCard(inputValues)
+    .then((newCard) => {
+      cardsList.prepend(getCardElement(newCard));
+      addPostForm.reset();
+      resetValidation(addPostForm, settings);
+      disableButton(cardSubmitBtn, settings);
+      closeModal(addModal);
+    })
+    .catch(console.error)
+    .finally(() => setButtonText(submitBtn, false));
 }
 
-addPostForm.addEventListener("submit", handleAddCardSubmit);
+// const cardElement = getCardElement(inputValues);
+// cardsList.prepend(cardElement);
+
+// addPostForm.reset();
+// resetValidation(addPostForm, settings);
+// disableButton(cardSubmitBtn); //added*****************
+// closeModal(addModal);
 
 //looping through cards
-initialCards.forEach(function (item) {
-  const cardElement = getCardElement(item);
-  cardsList.append(cardElement);
-});
+// initialCards.forEach(function (item) {
+//   const cardElement = getCardElement(item);
+//   cardsList.append(cardElement);
+// });
 
 modalPreviewClose.addEventListener("click", () => {
   closeModal(modalPreview);
 });
 
+avatarBtn.addEventListener("click", () => {
+  resetValidation(avatarForm, settings);
+  openModal(avatarModal);
+});
+
+avatarCloseBtn.addEventListener("click", () => closeModal(avatarModal));
+
+avatarForm.addEventListener("submit", (e) => {
+  e.preventDefault();
+  const submitBtn = e.submitter;
+  const url = avatarForm.querySelector("#profile-avatar-input").value.trim();
+
+  setButtonText(submitBtn, true, "Saving...");
+  api
+    .updateAvatar(url)
+    .then((user) => {
+      document.getElementById("profile-avatar").src = user.avatar;
```

Comment:
`document.getElementById("profile-avatar")`

It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine